### PR TITLE
Stop XMP parsing on non-well-formed XML

### PR DESCRIPTION
**Pull request by mattiasw**
#240

### CODE REVIEW

**Review by karfau — APPROVED** [review]
Looks great.
Some people take the following approach for the non spec compliant constructor args: the native implementation (and polyfills) will ignore them, so they just pass them always.
But in your case it's very clear where those arguments are applicable.

**Comment by jseagull** [comment]
The issue has been resolved. However I have a question why the whole XML in split into two parts by `extractCompleteChunks`. That's where the non-well-formed XML come from.

**Comment by mattiasw** [comment]
It's the program that wrote the file that split the XML in two. When the XML is too big to be stored in a single chunk in the JPEG file (64K) it has to be split into several chunks, each a maximum of 64K. The XMP specification says the first and main portion, a.k.a. StandardXMP, and the ExtendedXMP (all of the other chunks concatenated) should each be a a fully formed XMP metadata tree. The file in the issue does not follow this rule, the main portion is prematurely ended and therefore cannot be parsed.

That being said, I wonder if it would cause any issues if the parser tries to concatenate the StandardXMP with the ExtendedXMP in case parsing them separately fails. I will try and see if I can get that to work.

**Comment by mattiasw** [comment]
@jseagull Please have a look now. It seems to work fine. The downside is it will take longer for files where the XML is really broken and a concatenation does not help.

File: src/xmp-tags.js [hunk 1]
```diff
@@ -54,20 +75,20 @@ function readTags(tags, chunkDataView) {
         tags._raw = (tags._raw || '') + raw;
         const rdf = getRDF(doc);
 
-        return objectAssign(tags, parseXMPObject(convertToObject(rdf, true)));
+        objectAssign(tags, parseXMPObject(convertToObject(rdf, true)));
     } catch (error) {
-        return tags;
+        return true;
     }
+    return false;
```
Comment (karfau) [inline src/xmp-tags.js:82]: That looks a bit weird for my taste, a bit like bash, where exit code 0 is marking a success...

Comment (mattiasw) [reply src/xmp-tags.js:82]: Yeah, it didn't feel perfect. It made the code simpler than the first version that used exceptions but I will give it some more thought.

Comment (mattiasw) [reply src/xmp-tags.js]: I just flipped it around.

**Comment by jseagull** [comment]
LGTM

**Pull request merged**